route_builder.go

```diff
@@ -24,6 +24,7 @@ type RouteBuilder struct {
 	httpMethod  string        // required
 	function    RouteFunction // required
 	filters     []FilterFunction
+	iffunctions []RouteSelectionConditionFunction
```

Comment:
I'm thinking I should rename this to `conditions`; it won't change the external interface but seems like it would read better.